js/actionCreators/files.js

```diff
@@ -236,14 +229,35 @@ export function fetchMediaTags(file, id) {
       .then(data => {
         // There's more data here, but we don't have a use for it yet:
         // https://github.com/aadsm/jsmediatags#shortcuts
-        const { artist, title, picture } = data.tags;
+        const { artist, title, picture } = data.common;
         let albumArtUrl = null;
-        if (picture) {
-          const byteArray = new Uint8Array(picture.data);
-          const blob = new Blob([byteArray], { type: picture.type });
+        if (picture && picture.length >= 1) {
+          const byteArray = new Uint8Array(picture[0].data);
+          const blob = new Blob([byteArray], { type: picture[0].format });
           albumArtUrl = URL.createObjectURL(blob);
         }
         dispatch({ type: SET_MEDIA_TAGS, artist, title, albumArtUrl, id });
+
+        // If we're not going to load this right away,
+        // we should set duration on our own
+        if (data.format.duration != null) {
+          dispatch({
+            type: SET_MEDIA_DURATION,
+            duration: data.format.duration,
+            id
+          });
+        } else {
+          //  ToDo? dispatch(fetchMediaDuration(canonicalUrl, id));
```

Comment:
Are there some parsers which can't get duration?

Reply:
I don't think there are any, but there maybe specific case that is not possible. Anyway, probably good to make use of the expensive fall back scenario, using the audio element.